### ISSUE TITLE
Support for Hyper-V platform

### ISSUE THREAD

**Issue by mlacko64** [aws/eks-anywhere#6249]
**What would you like to be added**:

Support for Hyper-V platform.

**Why is this needed**:

Currently , only baremetal deployment using Ubuntu image is possible. Baremetal deployment using Bottlerocket will fail - I have opened also feature request to Bottlerocket , so it does support at least baremetal installation, see [here](https://github.com/bottlerocket-os/bottlerocket/issues/3277). Still full native Hyper-V support will be nice too.
Thank you very much

**Comment by chrisdoherty4** [comment]
@mlacko64 could you explain a little more about your environment and what you're trying to do?

Hyper-V is a hypervisor so deploying to a Hyper-V environment would fall outside the scope of our bare metal offering.

**Comment by mlacko64** [comment]
Hello @chrisdoherty4 ,
yes it is hypervisor, but limiting baremetal installation to physical HW is also limiting EKS-A deployment itself (possibilities).

Currently baremetal is officially supported on physical servers only, maybe the reason is BMC. But in reality you can do baremetal installation on VMWare workstation for example (because Bottlerocket is able to run on VMWare) and do also EKS-A upgrade etc; just, you have to to do BMC tasks manually.

Reason why to run it on hypervisor is development. You can run production on physical server with support, etc. That is fine. But with having possibilities to install it in into hypervisor, you can simplify your development and testing (for example testing upgrade to new version of EKS-A).

So in short, if some native Hyper-V deployment will be possible (like currently is on vSphere) that will be of course best, but even just if BottleRocket is able to boot and run normally on Hyper-V, will be enough this (my) case, so I can run development of baremetal environment on virtual machines.

This is similar for example to OpenShift , you can do baremetal installation on any HW, even virtual one (VMWare or Hyper-V).

**Comment by chrisdoherty4** [comment]
Hi @mlacko64. 

Thank you for the feedback. EKS Anywhere defines bare metal as physical hardware only. We think there are better solutions to provisioning on virtualization platforms such as Hyper-V or VMWare. For example, in VMWares case the underlying technology EKS Anywhere utilizes is cluster-api-provider-vsphere which provides better support for vSphere specific features than the technology used for bare metal deployments ever could.

We don't currently have Hyper-V on our roadmap but I've forwarded your feedback to the product management team for consideration.

Thanks again.

**Comment by zjpiazza** [comment]
Oh man I really need Hyper-V support, any changes to this discussion since 2023?
